runtime/cudaq/utils/extension_point.h

```diff
@@ -0,0 +1,168 @@
+/****************************************************************-*- C++ -*-****
+ * Copyright (c) 2024 NVIDIA Corporation & Affiliates.                         *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <unordered_map>
```

Comment:
I think nvcc needed this change:
```suggestion
#include <stdexcept>
#include <unordered_map>
```